gobblin-metrics/src/test/java/gobblin/metrics/performance/PerformanceUtils.java

```diff
@@ -0,0 +1,204 @@
+    for(Map.Entry<List<Integer>, Double> result : results.entrySet()) {
+      List<Integer> p = result.getKey();
+      System.out.println(String
+          .format("%d\t%d\t%d\t%d\t%d\t%d\t%d\t%f", p.get(0), p.get(1), p.get(2), p.get(3), p.get(4), p.get(5),
+              p.get(6), result.getValue()));
+    }
+  }
+
+  /**
+   * Runs a single performance test. Creates a {@link gobblin.metrics.MetricContext} tree, spawns a number of threads,
+   * uses and {@link Incrementer} to update the metrics repeatedly, then determines the achieved QPS in number
+   * of iterations of {@link Incrementer} per second.
+   *
+   * @param threads Number of threads to spawn. Each thread will have an {@link Incrementer} and update metrics.
+   * @param depth Depth of the {@link gobblin.metrics.MetricContext} tree.
+   * @param forkAtDepth If multiple threads, each thread has its own {@link gobblin.metrics.MetricContext}. This
+   *                    parameter sets the first level in the tree where the per-thread MetricContexts branch off.
+   * @param counters Number of counters to generate per thread.
+   * @param meters Number of meters to generate per thread.
+   * @param histograms Number of histograms to generate per thread.
+   * @param timers Number of timers to generate per thread.
+   * @param queries Number of increments to do, divided among all threads.
+   * @return total QPS achieved (e.g. total increments per second in the {@link Incrementer}s)
+   * @throws Exception
+   */
+  @Builder(buildMethodName = "run", builderMethodName = "singleTest")
+  public static double _singleTest(int threads, int depth, int forkAtDepth, int counters, int meters, int histograms,
+      int timers, long queries) throws Exception {
+
+    System.gc();
+
+    ExecutorService executorService = Executors.newFixedThreadPool(threads);
+    if(queries == 0) {
+      queries = 50000000l;
+    }
+    long queriesPerThread = queries / threads;
+    long actualQueries = queriesPerThread * threads;
+
+    MetricsUpdater commonUpdater = MetricsUpdater.builder().depth(forkAtDepth).build();
+
+    List<Incrementer> incrementerList = Lists.newArrayList();
+    while(incrementerList.size() < threads) {
+      final MetricsUpdater metricsUpdater = MetricsUpdater.builder().baseContext(commonUpdater.getContext()).
+          depth(depth - forkAtDepth).counters(counters).meters(meters).histograms(histograms).timers(timers).build();
+      incrementerList.add(new Incrementer(queriesPerThread, new Runnable() {
+        @Override public void run() {
+          metricsUpdater.run();
+        }
+      }));
+    }
+
+    List<Future<Long>> incrementerFutures = Lists.newArrayList();
+    for(Incrementer incrementer : incrementerList) {
+      incrementerFutures.add(executorService.submit(incrementer));
+    }
+
+    long totalTime = 0;
+    for(Future<Long> future : incrementerFutures) {
+      log.info("Waiting for future.");
```

Comment:
Suggest removing this or making it more descriptive so it logs which some more metadata about the Future it is waiting on.